[ILM] [DOCS] add general info about steps (#36081)
elastic/elasticsearch: docs/reference/ilm/policy-definitions.asciidoc

The previous phase's actions must complete before {ILM} will check `min_age` and
transition into the next phase.

=== Phase Execution

The current phase definition, of an index's policy being executed, is stored
in the index's metadata. The phase and its actions are compiled into a series
of discrete steps that are executed. These steps can be thought of as distinct

> API calls to Elasticsearch. Since some {ILM} actions are more complex and involve

Reviewer: I don't think we should encourage users to think of each step as an API call as its not true and could leads to confusion. I wonder if we can word this differently?

Author: I was iffy about it too, the sentence after mentions how actions are complex and need to be broken down, so maybe the sentence can simply be deleted

Author: deleted it

> multiple actions against an index, each of these is done in its own step. This

Reviewer: ```suggestion
multiple actions against an index, each of these are done in their own step. This
```

The previous phase's actions must complete before {ILM} will check `min_age` and
transition into the next phase.

=== Phase Execution

beta[]

The current phase definition, of an index's policy being executed, is stored
in the index's metadata. The phase and its actions are compiled into a series
of discrete steps that are executed. Since some {ILM} actions are more complex

> and involve multiple actions against an index, each of these are done in their

Reviewer: ```suggestion
and involve multiple operations against an index, each of these operations are
```
I wonder if we should avoid overloading the term "action" here since action already means something int he context of ILM?

> own step. This provides better visibility into the execution of the phases. The

Reviewer: ```suggestion
done in isolation in a unit called a step. This provides better visibility into
the execution of the phases. The
```

Reviewer: continuation of suggestion form above since Github's suggest feature only allows you to provide single line suggestions

> own step. This provides better visibility into the execution of the phases. The

Reviewer: ` This provides better visibility into the execution of the phases` -> I'm not sure about this sentence since this is not actually the reason we have steps (Though it is the reason why we expose steps in the ILM explain API). The reason we split the actions into steps is so we have better control over the execution of the lifecycle and so we know exactly which operations have been executed and which are still to be done.

Author: another sentence that is unnecessary
